lesson6

KochevniKXXI opened the pull request.

KochevniKXXI committed:
Шестой урок

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/java/hh/school/lesson_6_zemskov/MainActivity.kt:
```diff
@@ -0,0 +1,86 @@
+        setContentView(binding.root)
+
+        val navigationTopLevel: NavigationBarView = binding.navigationTopLevel
+        binding.toolbar.title =
```
легче в каждом фрагменте иметь свой тулбар

KochevniKXXI replied:
Рассматривал этот вариант. Но так как тубар везде типовой решил попробовать таким образом.

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/java/hh/school/lesson_6_zemskov/MainActivity.kt:
```diff
@@ -0,0 +1,86 @@
+        }
+    }
+
+    fun addMenu(
```
такое надо закрывать интерфейсом, а не обращаться к MainActivity, вдруг фрагменты будут лежать в другой, ну а вообще лучше делать тулбары во фрагментах

KochevniKXXI replied:
Про интерфейс понял.

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/java/hh/school/lesson_6_zemskov/PageFragment.kt:
```diff
@@ -0,0 +1,61 @@
+    private var _binding: FragmentPageBinding? = null
+    private val binding get() = _binding!!
+
+    private var paramUrl: String? = null
```
а можно `private val paramUrl by lazy { arguments.getString(ARG_URL) }`

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/java/hh/school/lesson_6_zemskov/PageFragment.kt:
```diff
@@ -0,0 +1,61 @@
+        @JvmStatic
+        fun newInstance(paramUrl: String, paramCaption: String) =
+            PageFragment().apply {
+                arguments = Bundle().apply {
```
не ошибка, есть еще конструктор bundleOf

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/java/hh/school/lesson_6_zemskov/ThreeItemFragment.kt:
```diff
@@ -0,0 +1,44 @@
+        binding.buttonShowBanner.setOnClickListener {
+            childFragmentManager.commit {
+                setReorderingAllowed(true)
+                (it as MaterialButton).text = if (binding.fragmentContainerViewPictures.isEmpty()) {
```
зачем it as 
легче уж binding.buttonShowBanner

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/java/hh/school/lesson_6_zemskov/adapters/ServiceItemDecoration.kt:
```diff
@@ -0,0 +1,26 @@
+    ) {
+        super.getItemOffsets(outRect, view, parent, state)
+
+        val space = parent.context.resources.getDimensionPixelOffset(dimenResId)
```
лучше в конструктор пиксели подавать, чтобы каждый раз это не резолвить через ресурсы

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/java/hh/school/lesson_6_zemskov/adapters/ServicesAdapter.kt:
```diff
@@ -0,0 +1,34 @@
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServiceViewHolder {
+        return ServiceViewHolder(
+            ItemServiceBinding.inflate(
```
тебе не нравится как я показывал ? подавать только parent внутрь
вьюха ведь привязана к самому вью холдеру
представь, что ты его будешь переиспользовать в другом адаптере, тебе придется это закопипастить

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/java/hh/school/lesson_6_zemskov/model/Service.kt:
```diff
@@ -0,0 +1,27 @@
+sealed interface Service {
+    val name: String
+    val serialNumber: String
+    @get:DrawableRes val iconResId: Int
```
зачем get: ?

KochevniKXXI replied:
Без него ошибка была. Перепроверю

KochevniKXXI replied:
да, ошибка. This annotation is not applicable to target 'member property without backing field or delegate'

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/res/layout/fragment_page.xml:
```diff
@@ -0,0 +1,31 @@
+    tools:context=".PageFragment"
+    android:orientation="vertical">
+
+    <com.google.android.material.card.MaterialCardView
```
а зачем карточка ?

DmitriyTSt replied:
для клика, понял

KochevniKXXI replied:
Да, чтобы с анимацией состояний не заморачиваться

DmitriyTSt commented on Lesson_6_Zemskov/app/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,23 @@
+    <string name="item_lamp_title">Фонарик</string>
+    <string name="item_service_one_edit_text_hint">Новые показания</string>
+    <string name="alert_message">Необходимо подать показания до %s</string>
+    <string name="helper_message">Показания сданы &lt;b>%s&lt;/b> и будут учтены при следующем расчете &lt;b>%s&lt;/b></string>
```
так как ты это форматом вставляешь, то наверное можно еще присмотреться к SpannableString, html это такой, крайний вариант, не особо нативный )

KochevniKXXI replied:
Форматирование spannable и annotation тоже смотрел

KochevniKXXI committed:
Шестой урок.

Исправлены ошибки.
Добавлен NavigationRail для широких экранов.
Добавлена прокрутка для третьего экрана.

KochevniKXXI committed:
Шестой урок.

Исправлены ошибки.
Добавлен NavigationRail для широких экранов.
Добавлена прокрутка для третьего экрана.
